bf luis:convert: Error of "Roles must be unique across entity types." triggers by having multiple enabledForAllModels

## Versions
@microsoft/botframework-cli/4.8.0 win32-x64 node-v12.16.3
Windows 10

## Describe the bug
Having more than 1 phrase list declaration with enabledForAllModels annotation will cause the CLI errors out.

> [ERROR] line 7:0 - line 7:60: Roles must be unique across entity types. Invalid role definition found "my_gpl(interchangeable)". Prior definition - '@ phraselist Iam_gpl(interchangeable) hasRoles enabledForAllModels'

## To Reproduce
```
@ phraselist Iam_gpl(interchangeable) enabledForAllModels =
	- I
	- I'm
	- I am
	- I be

@ phraselist my_gpl(interchangeable) enabledForAllModels =
	- my
	- mine
	- me
```

--- a/packages/lu/src/parser/lufile/parseFileContents.js
+++ b/packages/lu/src/parser/lufile/parseFileContents.js
@@ -1015,7 +1015,7 @@ const validateAndGetRoles = function(parsedContent, roles, line, entityName, ent
             let roleFound = parsedContent.LUISJsonStructure.flatListOfEntityAndRoles.find(item => item.roles.includes(role) || item.name === role);
             if (roleFound !== undefined) {
                 // PL entities use roles for things like interchangeable, disabled, enabled for all models. There are really not 'dupes'.
-                let hasBadNonPLRoles = (roleFound.roles || []).filter(item => item !== PLCONSTS.INTERCHANGEABLE && item !== PLCONSTS.ENABLEDFORALLMODELS && item !== PLCONSTS.DISABLED);
+                let hasBadNonPLRoles = (roleFound.roles || []).filter(item => item.toLowerCase() !== PLCONSTS.INTERCHANGEABLE && item.toLowerCase() !== PLCONSTS.ENABLEDFORALLMODELS && item.toLowerCase() !== PLCONSTS.DISABLED);
                 if (hasBadNonPLRoles.length !== 0) {
                     let errorMsg = `Roles must be unique across entity types. Invalid role definition found "${entityName}". Prior definition - '@ ${roleFound.type} ${roleFound.name}${roleFound.roles.length > 0 ? ` hasRoles ${roleFound.roles.join(',')}` : ``}'`;
                     let error = BuildDiagnostic({

--- a/packages/lu/src/parser/luis/luConverter.js
+++ b/packages/lu/src/parser/luis/luConverter.js
@@ -456,7 +456,7 @@ const addRolesAndFeatures = function(entity) {
 const updateUtterancesList = function (srcCollection, tgtCollection, attribute) {
     (srcCollection || []).forEach(srcItem => {
         let matchInTarget = tgtCollection.find(item => item.intent.name == srcItem.intent);
-        if(matchInTarget.utterances.length === 0) {
+        if(!matchInTarget || matchInTarget.utterances.length === 0) {
             addUtteranceToCollection(attribute, srcItem, matchInTarget);
             return;
         }

--- a/packages/lu/test/parser/lufile/parseFileContents.NewEntityDefinition.test.js
+++ b/packages/lu/test/parser/lufile/parseFileContents.NewEntityDefinition.test.js
@@ -1175,5 +1175,27 @@ describe('V2 Entity definitions using @ notation', function () {
                 .catch(err => done(err))
         })
     })
+
+    describe('phrase lists with enabledForAllModels', function(){
+        it('BF-CLI #789 - phrase lists with enabledForAllModels parse correctly', function(done){
+            let luContent1 = new luObj(`@ phraselist Iam_gpl(interchangeable) enabledForAllModels =
+            - I
+            - I'm
+            - I am
+            - I be
+        
+        @ phraselist my_gpl(interchangeable) enabledForAllModels =
+            - my
+            - mine
+            - me
+            `);
+            luBuilder.fromLUAsync([luContent1])
+                .then(res => {
+                    assert.isTrue(res.model_features.length === 2);
+                    done()
+                })
+                .catch(err => done(err))
+        })
+    })
     
 });

